eva-accession-ws/src/main/java/uk/ac/ebi/eva/accession/ws/rest/ClusteredVariantsRestController.java

```diff
@@ -72,12 +81,39 @@ public ClusteredVariantsRestController(
             + "variants (RS) represented by the given identifier. For a description of the response, see "
             + "https://github.com/EBIvariation/eva-accession/wiki/Import-accessions-from-dbSNP#clustered-variant-refsnp-or-rs")
     @GetMapping(value = "/{identifier}", produces = "application/json")
-    public List<AccessionResponseDTO<ClusteredVariant, IClusteredVariant, String, Long>> get(
+    public ResponseEntity<List<AccessionResponseDTO<ClusteredVariant, IClusteredVariant, String, Long>>> get(
             @PathVariable @ApiParam(value = "Numerical identifier of a clustered variant, e.g.: 3000000000",
                                     required = true) Long identifier)
-            throws AccessionMergedException, AccessionDoesNotExistException, AccessionDeprecatedException {
+            throws AccessionMergedException, AccessionDoesNotExistException {
+        try {
+            return ResponseEntity.ok(Collections.singletonList(basicRestController.get(identifier)));
+        } catch (AccessionDeprecatedException e) {
+            // not done with an exception handler because the only way to get the accession parameter would be parsing
+            // the exception message
+            return ResponseEntity.status(HttpStatus.GONE).body(getDeprecatedClusteredVariant(identifier));
+        }
+    }
+
+    /**
+     * Retrieve the information in the collection for inactive objects.
+     *
+     * This method is necessary because the behaviour of BasicRestController is to return the HttpStatus.GONE with an
+     * error exception message in the body. We want instead to return the HttpStatus.GONE with the variant in the body
+     */
+    private List<AccessionResponseDTO<ClusteredVariant, IClusteredVariant, String, Long>> getDeprecatedClusteredVariant(
+            Long identifier) {
+        IAccessionedObject<IClusteredVariant, ?, Long> accessionedObjectWrongType = inactiveService
```

Comment:
```
IEvent<MODEL, ACCESSION> {
...
    List<? extends IAccessionedObject<MODEL, ?, ACCESSION>> getInactiveObjects();
```
that question mark represents the hash type, and it can't be a template parameter unless the IEvent class depends on that parameter, and that would be too much coupling. The events don't need to know the hash type ( which is included in the MODEL for eva-accession but it's not required at accession-commons level)